pkg/controller/daemon/daemon_controller_test.go

```diff
@@ -1830,7 +1830,7 @@ func TestNodeShouldRunDaemonPod(t *testing.T) {
 			if shouldContinueRunning != c.shouldContinueRunning {
 				t.Errorf("[%v] strategy: %v, predicateName: %v expected shouldContinueRunning: %v, got: %v", i, c.ds.Spec.UpdateStrategy.Type, c.predicateName, c.shouldContinueRunning, shouldContinueRunning)
 			}
-			if err != c.err {
+			if err.Error() != c.err.Error() {
```

Comment:
@mikedanese 
we could compare two error object directly. They could not be equal, even they have the same error messages.